test/e2e/upgrade_test.go

```diff
@@ -47,6 +47,16 @@ func runUpgradeFlowWithCheckpoint(test *framework.ClusterE2ETest, updateVersion
 	test.DeleteCluster()
 }
 
+func runSimpleUpgradeFlowForBareMetal(test *framework.ClusterE2ETest, updateVersion v1alpha1.KubernetesVersion, clusterOpts ...framework.ClusterE2ETestOpt) {
+	test.GenerateClusterConfig()
+	test.GenerateHardwareConfig()
+	test.CreateCluster()
+	test.UpgradeCluster(clusterOpts)
+	test.ValidateCluster(updateVersion)
+	test.StopIfFailed()
+	test.DeleteCluster()
```

Comment:
I recommend calling test.ValidateHardwareDecommissioned() after DeleteCluster() to ensure delete is successful post upgrade.

Reply:
Also just FYI, we have test.PowerOnHardware() and test.PowerOffHardware() if you want to set the hardware power state prior to create and/or upgrade.